### PR TITLE
Scale the About dialog box based on screen size - Needs Multiple testers

### LINKED ISSUE #28789: Mantid about screen is tiny
When I start up Mantid the about screen is too small to read. The mouse cursor changes to resize it, but I cannot alter the size in a meaningful way. 

image

### PR DESCRIPTION
**Pull request by NickDraper**
**Problem:**
The about box could appear tiny on high res small laptop screens.

**Description of work.**
see commits.
TLDR:
refactored the About dialog box into code from QI file and made it scale based on the screen it appears on.

**To test:**
This should ideally be tested on (stick a screenshot in the comments, with os & screen resolution):
- [x] Windows
- [x] Ubuntu
- [x] RHEL or CENTOS
- [x] Mac
- [x] A high res laptop screen

1. open Mantid and help->about if it does not appear automatically
1. If you have more than one screen check it with Mantid on both at the time the help-about is clicked
1. check the links etc still work.

Windows 2560 * 1440
image

Windows 1920 * 1080
image

fixed pixel size .ui based version before these changes
image

Fixes #28789

*This does not require release notes* because **the dialog is new for this release**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by ConorMFinn** [comment]
```
Traceback (most recent call last):
  File "/home/conor/repos/mantid-review/qt/applications/workbench/workbench/app/mainwindow.py", line 641, in open_about
    about = AboutPresenter(self)
  File "/home/conor/repos/mantid-review/qt/applications/workbench/workbench/widgets/about/presenter.py", line 28, in __init__
    self.view = view if view else AboutView(parent, self, version_str(), release_date().strip())
  File "/home/conor/repos/mantid-review/qt/applications/workbench/workbench/widgets/about/view.py", line 37, in __init__
    self.setupUI()
  File "/home/conor/repos/mantid-review/qt/applications/workbench/workbench/widgets/about/view.py", line 68, in setupUI
    screen = QGuiApplication.screenAt(self.parent().geometry().center())
AttributeError: type object 'QGuiApplication' has no attribute 'screenAt'
```
Getting this error on Ubuntu 18.04

**Comment by dtasev** [comment]
I tried to rebuild the Windows build with a package but it seems a test assertion has failed in it

```
10:42:07 2500/2550 Test #2230: workbench.test_plot_settings.test_plot_settings ..............................................................................***Failed    3.05 sec
10:42:07 ............F
10:42:07 ======================================================================
10:42:07 FAIL: test_load_current_setting_values (test_plot_settings.PlotsSettingsTest)
10:42:07 ----------------------------------------------------------------------
10:42:07 Traceback (most recent call last):
10:42:07   File "C:\Jenkins\workspace\pull_requests-windows\external\src\ThirdParty\lib\python3.8\lib\unittest\mock.py", line 1342, in patched
10:42:07     return func(*newargs, **newkeywargs)
10:42:07   File "C:/Jenkins/workspace/pull_requests-windows/qt/applications/workbench/workbench/widgets/settings/plots/test/test_plot_settings.py", line 36, in test_load_current_setting_values
10:42:07     mock_ConfigService.getString.assert_has_calls([call(PlotSettings.NORMALIZATION),
10:42:07   File "C:\Jenkins\workspace\pull_requests-windows\external\src\ThirdParty\lib\python3.8\lib\unittest\mock.py", line 944, in assert_has_calls
10:42:07     raise AssertionError(
10:42:07 AssertionError: Calls not found.
10:42:07 Expected: [call('graph1d.autodistribution'),
10:42:07  call('plots.ShowTitle'),
10:42:07  call('plots.xAxesScale'),
10:42:07  call('plots.yAxesScale'),
10:42:07  call('plots.line.Style'),
10:42:07  call('plots.line.Width'),
10:42:07  call('plots.marker.Style'),
10:42:07  call('plots.marker.Size'),
10:42:07  call('plots.errorbar.Width'),
10:42:07  call('plots.errorbar.Capsize'),
10:42:07  call('plots.errorbar.CapThickness'),
10:42:07  call('plots.errorbar.errorEvery')]
10:42:07 Actual: [call('graph1d.autodistribution'),
10:42:07  call('plots.ShowTitle'),
10:42:07  call('plots.xAxesScale'),
10:42:07  call('plots.yAxesScale'),
10:42:07  call('plots.line.Style'),
10:42:07  call('plots.line.DrawStyle'),
10:42:07  call('plots.line.Width'),
10:42:07  call('plots.marker.Style'),
10:42:07  call('plots.marker.Size'),
10:42:07  call('plots.errorbar.Width'),
10:42:07  call('plots.errorbar.Capsize'),
10:42:07  call('plots.errorbar.CapThickness'),
10:42:07  call('plots.errorbar.errorEvery')]
10:42:07 
10:42:07 ----------------------------------------------------------------------
10:42:07 Ran 13 tests in 0.985s
10:42:07 
10:42:07 FAILED (failures=1)
```
from https://builds.mantidproject.org/job/pull_requests-windows/36655/console

**Comment by NickDraper** [comment]
> ```
> Traceback (most recent call last):
>   File "/home/conor/repos/mantid-review/qt/applications/workbench/workbench/app/mainwindow.py", line 641, in open_about
>     about = AboutPresenter(self)
>   File "/home/conor/repos/mantid-review/qt/applications/workbench/workbench/widgets/about/presenter.py", line 28, in __init__
>     self.view = view if view else AboutView(parent, self, version_str(), release_date().strip())
>   File "/home/conor/repos/mantid-review/qt/applications/workbench/workbench/widgets/about/view.py", line 37, in __init__
>     self.setupUI()
>   File "/home/conor/repos/mantid-review/qt/applications/workbench/workbench/widgets/about/view.py", line 68, in setupUI
>     screen = QGuiApplication.screenAt(self.parent().geometry().center())
> AttributeError: type object 'QGuiApplication' has no attribute 'screenAt'
> ```
> 
> Getting this error on Ubuntu 18.04 when I attempt to open the About Mantid Workbench window.

Ah, that method was only introduced in 5.10, I'll change what I'm using.

**Comment by NickDraper** [comment]
> > ```
> > Traceback (most recent call last):
> >   File "/home/conor/repos/mantid-review/qt/applications/workbench/workbench/app/mainwindow.py", line 641, in open_about
> >     about = AboutPresenter(self)
> >   File "/home/conor/repos/mantid-review/qt/applications/workbench/workbench/widgets/about/presenter.py", line 28, in __init__
> >     self.view = view if view else AboutView(parent, self, version_str(), release_date().strip())
> >   File "/home/conor/repos/mantid-review/qt/applications/workbench/workbench/widgets/about/view.py", line 37, in __init__
> >     self.setupUI()
> >   File "/home/conor/repos/mantid-review/qt/applications/workbench/workbench/widgets/about/view.py", line 68, in setupUI
> >     screen = QGuiApplication.screenAt(self.parent().geometry().center())
> > AttributeError: type object 'QGuiApplication' has no attribute 'screenAt'
> > ```
> > 
> > 
> > Getting this error on Ubuntu 18.04 when I attempt to open the About Mantid Workbench window.
> 
> Ah, that method was only introduced in 5.10, I'll change what I'm using.

**Comment by NickDraper** [comment]
> I tried to rebuild the Windows build with a package but it seems a test assertion has failed in it
> 
> ```
> 10:42:07 2500/2550 Test #2230: workbench.test_plot_settings.test_plot_settings ..............................................................................***Failed    3.05 sec
> from https://builds.mantidproject.org/job/pull_requests-windows/36655/console


This is a bug that got onto master, it's been fixed now, I'll merge it in.

**Comment by martyngigg** [comment]
I was just testing locally and the "Do not show again until next release" checkbox does not seem to work. I tick it, close workbench then start it up and the about screen appears again

**Comment by AnthonyLim23** [comment]
Just tried on Windows with a 2840 x 2160 screen.

image

Closing the window causes errors
```
Looking for  at ISIS.
Blank instrument specified, using default instrument of HIFI.
Instrument 'HIFI' found as HIFI at ISIS.
Determining ioRaw buffer size

Unable to find loadraw.readbuffer.size in the properties file
loadraw.readbuffer.size not found, setting to 200000
Determining ioRaw buffer size

Unable to find loadraw.readbuffer.size in the properties file
loadraw.readbuffer.size not found, setting to 200000
Not adding gui "Frequency_Domain_Analysis_Old.py"
Looking for  at ISIS.
Blank instrument specified, using default instrument of HIFI.
Instrument 'HIFI' found as HIFI at ISIS.
Determining ioRaw buffer size

Unable to find loadraw.readbuffer.size in the properties file
loadraw.readbuffer.size not found, setting to 200000
Determining ioRaw buffer size

Unable to find loadraw.readbuffer.size in the properties file
loadraw.readbuffer.size not found, setting to 200000
```

**Comment by martyngigg** [comment]
On my CentOS VM (the closest to IDAaaS I can get) I see:

MantidAboutDialogCentOS-VM-1776x944

This is the same as master on RHEL with the gray boxes and large text. I'm not sure it's overly worth spending too much time on it?

**Comment by martyngigg** [comment]
> ```
> Looking for  at ISIS.
> Blank instrument specified, using default instrument of HIFI.
> Instrument 'HIFI' found as HIFI at ISIS.
> Determining ioRaw buffer size
> 
> Unable to find loadraw.readbuffer.size in the properties file
> loadraw.readbuffer.size not found, setting to 200000
> Determining ioRaw buffer size
> 
> Unable to find loadraw.readbuffer.size in the properties file
> loadraw.readbuffer.size not found, setting to 200000
> Not adding gui "Frequency_Domain_Analysis_Old.py"
> Looking for  at ISIS.
> Blank instrument specified, using default instrument of HIFI.
> Instrument 'HIFI' found as HIFI at ISIS.
> Determining ioRaw buffer size
> 
> Unable to find loadraw.readbuffer.size in the properties file
> loadraw.readbuffer.size not found, setting to 200000
> Determining ioRaw buffer size
> 
> Unable to find loadraw.readbuffer.size in the properties file
> loadraw.readbuffer.size not found, setting to 200000
> ```

These are the debug messages you get when `LoadRaw` is constructed. I don't think they are errors but do new ones appear after the initial startup ones when you close the dialog?

**Comment by AnthonyLim23** [comment]
I opened Mantid and closed the about screen. I then noticed the errors. So I reopened the about screen, no errors. Then I closed it and more errors appeared.

**Comment by DanielMurphy22** [comment]
Mac: Mantid About looks good on both 2880 x 1800 Mac Laptop screen and 1920 x 1200 Monitor.
The same as @martyngigg saw, with ticking the "Do not show.." button and restarting, the About dialog still showed. Is it possible this is a build versus package difference?

The only odd thing on Mac is trying to find the Mantid About menu. Like Mantid Settings is under Python>Preferences, Mantid About has been put under ` About Python`. 
Screenshot 2020-06-16 at 15 54 55 (2)

**Comment by martyngigg** [comment]
> The only odd thing on Mac is trying to find the Mantid About menu. Like Mantid Settings is under Python>Preferences, Mantid About has been put under ` About Python`.
> Screenshot 2020-06-16 at 15 54 55 (2)
This is a Mac thing that we don't control, Qt puts it there as thats where macOS apps generally put it. We should probably look if we can change the name that appears next to the Apple icon as it's just currently picking up the executable name. We might have to play with the plist file but this should be a separate issue.

**Comment by NickDraper** [comment]
> > ```
> > Looking for  at ISIS.
> > Blank instrument specified, using default instrument of HIFI.
> > Instrument 'HIFI' found as HIFI at ISIS.
> > Determining ioRaw buffer size
> > 
> > Unable to find loadraw.readbuffer.size in the properties file
> > loadraw.readbuffer.size not found, setting to 200000
> > Determining ioRaw buffer size
> > 
> > Unable to find loadraw.readbuffer.size in the properties file
> > loadraw.readbuffer.size not found, setting to 200000
> > Not adding gui "Frequency_Domain_Analysis_Old.py"
> > Looking for  at ISIS.
> > Blank instrument specified, using default instrument of HIFI.
> > Instrument 'HIFI' found as HIFI at ISIS.
> > Determining ioRaw buffer size
> > 
> > Unable to find loadraw.readbuffer.size in the properties file
> > loadraw.readbuffer.size not found, setting to 200000
> > Determining ioRaw buffer size
> > 
> > Unable to find loadraw.readbuffer.size in the properties file
> > loadraw.readbuffer.size not found, setting to 200000
> > ```
> 
> These are the debug messages you get when `LoadRaw` is constructed. I don't think they are errors but do new ones appear after the initial startup ones when you close the dialog?

So these appear when calling the parent workbench app to let it know that the config may have been changed.
``` python
self.parent.config_updated()
```
This does 
``` python
    def config_updated(self):
        """
        Updates the widgets that depend on settings from the Workbench Config.
        """
        self.editor.load_settings_from_config(CONF)
        self.project.load_settings_from_config(CONF)
        self.algorithm_selector.refresh()
        self.populate_interfaces_menu()
        self.workspacewidget.refresh_workspaces()
```
the log messages are coming from these updates

**Comment by NickDraper** [comment]
> Mac: Mantid About looks good on both 2880 x 1800 Mac Laptop screen and 1920 x 1200 Monitor.
> The same as @martyngigg saw, with ticking the "Do not show.." button and restarting, the About dialog still showed. Is it possible this is a build versus package difference?
> 
> The only odd thing on Mac is trying to find the Mantid About menu. Like Mantid Settings is under Python>Preferences, Mantid About has been put under ` About Python`.
> Screenshot 2020-06-16 at 15 54 55 (2)

So this is a special mac feature,  it looks like there may be a way to change the bundle name, but it needs another python dependency.
https://stackoverflow.com/questions/5047734/in-osx-change-application-name-from-python

Created a seperate issue to look into this: #28861

File: qt/applications/workbench/workbench/widgets/about/view.py [hunk 1]
```diff
@@ -5,47 +5,327 @@
...
+    def determine_dialog_dimensions(self):
+        width = REFERENCE_WIDTH
+        height = REFERENCE_HEIGHT
+
+        # gets the width of the screen where the main window is
+        screen = None
+        if hasattr(QGuiApplication,"screenAt"):
+            screen = QGuiApplication.screenAt(self.parent().geometry().center())
+        else:
+            screen = QGuiApplication.prmaryScreen()
```
Comment (martyngigg) [inline qt/applications/workbench/workbench/widgets/about/view.py:66]: There's a typo here -> `QGuiApplication.primaryScreen()` that causes it to fail on startup.

**Comment by Harrietbrown** [comment]
the window rescales and looks fine on my regular 1920x1080 and when I change the resolution to 800x600. I noticed that all the other links open the page in a web browser except the release notes, should they all be the same?

**Comment by NickDraper** [comment]
> the window rescales and looks fine on my regular 1920x1080 and when I change the resolution to 800x600. I noticed that all the other links open the page in a web browser except the release notes, should they all be the same?

The links are as before, we can't have them all be the same as some are from the help and some are web pages.

**Comment by NickDraper** [comment]
@ConorMFinn @dtasev @DanielMurphy22 Could you give this another go now, please.

**Comment by DanielMurphy22** [comment]
The "Do Not Show" button is still not working on Mac.

**Comment by ConorMFinn** [comment]
Tested on Ubuntu at 1920x1080. 
- All buttons working.
- Size looks fine. 
- "Do Not Show" checkbox working fine.

Screenshot 2020-06-17 at 12 03 52

Looks good to me!

**Comment by dtasev** [comment]
Tested on Windows 10, 3840x2160 with 200% scaling.

- All buttons working.
- Size looks fine.
- "Do Not Show" checkbox working fine.

**Review by DanielMurphy22 — APPROVED** [review]
The "Do Not Show" tick box actually does work on Mac now I've retested. I seem to be the last one so speaking for the group, I think we're happy for this to go in!
